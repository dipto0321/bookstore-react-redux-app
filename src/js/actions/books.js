import uuid from 'uuid';
import {
  ADD_BOOK,
  REMOVE_BOOK,
  SWITCH_ADD_BOOK_MODE,
} from './actionTypes';


const addBook = ({ title, author, category }) => ({
  type: ADD_BOOK,
  book: {
    id: uuid(),
    title,
    author,
    category,
  },
});

const switchAddBookMode = () => ({
  type: SWITCH_ADD_BOOK_MODE,
});

const removeBook = id => ({
  type: REMOVE_BOOK,
  id,
});

export {
  addBook,
  removeBook,
  switchAddBookMode,
};
