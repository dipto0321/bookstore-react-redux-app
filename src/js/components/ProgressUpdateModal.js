import React from 'react';
import PropTypes from 'prop-types';
import Modal from 'react-modal';
import ProgressUpdateBody from './ProgressUpdateBody';

const ProgressUpdateModal = (
  {
    itemForProgressUpdate,
    progressUpdateMode,
    switchProgressUpdate,
  },
) => {
  const pagesArray = Array.from({ length: itemForProgressUpdate.page_count }, (v, k) => k + 1);
  return (
    <Modal
      isOpen={progressUpdateMode.on}
      ariaHideApp={false}
      closeTimeoutMS={200}
      onRequestClose={() => switchProgressUpdate()}
    >
      <div className="modal-controls">
        <button
          type="button"
          className="cancel-btn"
          onClick={() => switchProgressUpdate()}
        >
        X
        </button>
      </div>
      <div className="modal-body">
        <ProgressUpdateBody
          pagesArray={pagesArray}
          {...itemForProgressUpdate}
        />
      </div>
    </Modal>
  );
};

ProgressUpdateModal.propTypes = {
  itemForProgressUpdate: PropTypes.instanceOf(Object).isRequired,
  progressUpdateMode: PropTypes.instanceOf(Object).isRequired,
  switchProgressUpdate: PropTypes.func.isRequired,
};

export default ProgressUpdateModal;
