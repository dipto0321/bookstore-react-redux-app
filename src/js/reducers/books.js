import books from "../data/dummyBooks";

export default (state = books, action) => {
  switch (action.type) {
    case "ADD_BOOK":
      return [...state, action.book];
    case "REMOVE_BOOK":
      return state.filter(book => book.id !== action.id);
    default:
      return state;
  }
};
