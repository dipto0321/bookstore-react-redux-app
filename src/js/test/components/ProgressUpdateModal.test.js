import React from 'react';
import { shallow } from 'enzyme';
import ProgressUpdateModal from '../../components/ProgressUpdateModal';
import books from '../fixtures/books';

describe('ProgressUpdateModal component', () => {
  let wrapper;
  let switchProgressUpdate;

  beforeEach(() => {
    switchProgressUpdate = jest.fn();
    wrapper = shallow(
      <ProgressUpdateModal
        itemForProgressUpdate={books[2]}
        progressUpdateMode={{ on: false, id: '' }}
        switchProgressUpdate={switchProgressUpdate}
      />,
    );
  });

  test('should render correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  test('should call switchProgressUpdate', () => {
    wrapper.find('button').prop('onClick')();
    expect(switchProgressUpdate).toHaveBeenCalled();
  });
});
