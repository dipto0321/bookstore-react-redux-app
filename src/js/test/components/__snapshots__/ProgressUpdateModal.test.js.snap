// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProgressUpdateModal component should render correctly 1`] = `
<Modal
  ariaHideApp={false}
  bodyOpenClassName="ReactModal__Body--open"
  closeTimeoutMS={200}
  isOpen={false}
  onRequestClose={[Function]}
  parentSelector={[Function]}
  portalClassName="ReactModalPortal"
  role="dialog"
  shouldCloseOnEsc={true}
  shouldCloseOnOverlayClick={true}
  shouldFocusAfterRender={true}
  shouldReturnFocusAfterClose={true}
>
  <div
    className="modal-controls"
  >
    <button
      className="cancel-btn"
      onClick={[Function]}
      type="button"
    >
      X
    </button>
  </div>
  <div
    className="modal-body"
  >
    <ConnectFunction
      authors={
        Array [
          "Roddenberry",
        ]
      }
      book_id="2"
      category="Sci-Fi"
      chapters="30"
      currentChapter=""
      description="Nostrud aliquip velit eu ea amet qui exercitation anim mollit do minim commodo pariatur."
      id="2"
      imageLinks={
        Object {
          "smallThumbnail": "",
        }
      }
      item_id="two"
      pagesArray={Array []}
      title="Star Trek"
    />
  </div>
</Modal>
`;
